src/HtmPixelization.cc

```diff
@@ -0,0 +1,261 @@
+/*
+ * LSST Data Management System
+ * Copyright 2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// \file
+/// \brief This file contains the HtmPixelization class implementation.
+
+#include "lsst/sphgeom/HtmPixelization.h"
+
+#include "lsst/sphgeom/curve.h"
+#include "lsst/sphgeom/orientation.h"
+
+#include "PixelFinder.h"
+
+
+namespace lsst {
+namespace sphgeom {
+
+namespace {
+
+// Raw HTM root vertex coordinates. UnitVector3d objects are avoided so that
+// library clients cannot run into the static initialization order fiasco.
```

Comment:
Well, the answer to your question is that people should not instantiate global objects which do non-trivial stuff in their constructors like calling other complex libraries :smile_cat:

The problem is partially mitigated by shared libraries, before anyone can call any method in a shared library all its global objects have to be initialized. But if you do static linking then you are right, you either have to do static initialization or do crazy singleton-like stuff. In any case your solution is the best if you worry about clients doing nasty things. My unclear comment was that fiasco is avoided if everyone is not doing complex stuff in their globals constructors, your classes definitely qualify.